mystring.cpp

```diff
@@ -71,4 +71,40 @@ ostream & operator<< (ostream& os, const String &s)
 unsigned int String::length() const
 {  
     return size;
+}
+
+//overloading assigment operator
+String & String::operator=(const String &obj)
+{
+    //guard self assignment
+    if(this != &obj)
+    {
+        size = obj.size;
+        strcpy(text.get(), obj.text.get());
```

Comment:
i suggest using stcncpy here to control if the string is invalid

Reply:
ok, done